packages/story-editor/src/utils/getResourceBaseColor.js

```diff
@@ -46,16 +46,19 @@ function getImgNodeKey(elementId) {
   return `${IMG_NODE}_${elementId}`;
 }
 
-export function getMediaBaseColor(resource, onBaseColor) {
+export function getResourceBaseColor(resource) {
   const { type, src, poster } = resource;
-  setOrCreateImage({
-    src: type === 'video' ? poster : src,
+
+  const imageSrc = type === 'image' ? src : poster;
+
+  if (!imageSrc) {
+    return Promise.reject(new Error('No source to image'));
+  }
+  return setOrCreateImage({
```

Comment:
Nit: Maybe change the name of this now for readability to indicate that it's not just setting/creating image but also getting the average color?